cmake/configure/configure_1_cuda.cmake

```diff
@@ -24,80 +24,51 @@ SET(DEAL_II_WITH_CUDA FALSE CACHE BOOL "")
 
 MACRO(FEATURE_CUDA_FIND_EXTERNAL var)
 
-  #
-  # FIXME Restructure this call into a ./modules/FindCUDA.cmake file and
-  # use DEAL_II_PACKAGE_HANDLE
-  #
   FIND_PACKAGE(CUDA)
-  MARK_AS_ADVANCED(
-    CUDA_HOST_COMPILER
-    CUDA_SDK_ROOT_DIR
-    CUDA_TOOLKIT_ROOT_DIR
-    CUDA_USE_STATIC_CUDA_RUNTIME
-    )
 
   IF(CUDA_FOUND)
 
+    #
+    # CUDA was found, check whether we can actually use it:
+    #
     SET(${var} TRUE)
 
-    IF(DEFINED CUDA_DIR)
-      SET(CUDA_TOOLKIT_ROOT_DIR "${CUDA_DIR}")
+    IF("${CUDA_NVCC_FLAGS}" MATCHES "-arch[ ]*sm_([0-9]*)")
+      SET(CUDA_COMPUTE_CAPABILITY "${CMAKE_MATCH_1}")
+    ELSEIF("${CUDA_NVCC_FLAGS}" MATCHES "-arch=sm_([0-9]*)")
+      SET(CUDA_COMPUTE_CAPABILITY "${CMAKE_MATCH_1}")
+    ELSE()
+      #
+      # Assume a cuda compute capability of 35
+      #
+      SET(CUDA_COMPUTE_CAPABILITY "35")
```

Comment:
so if you can not detect the version you assume 35, which we support? That is confusing to me.